### PR TITLE
add tickets to operations

### PR DESCRIPTION
**Pull request by EduardoRFS**
## Problem

Users should be able to operate on tickets.

## Solution

This is solved through adding tickets to the ledger and to Transaction operation, because of this essentially the key for a ledger is a tuple between the `ticket` and wallet `address`.

## Related Issues

- #34

### CODE REVIEW

**Review by anchpop — APPROVED** [review]

File: protocol/ticket.re [hunk 1]
```diff
@@ -0,0 +1,28 @@
+open Tezos_interop;
+
+// TODO: I feel uneasy about this module
```
Comment (callistonianembrace) [inline protocol/ticket.re:3]: Why do you feel uneasy about this module?

Comment (callistonianembrace) [reply protocol/ticket.re:3]: I'm also reminded of functors like `Set_with_yojson_make` and `Map_with_yojson_make` - can they be unified?

**Review by callistonianembrace — APPROVED** [review]

File: protocol/ticket.re (hunk 1 above)
Comment (callistonianembrace) [reply protocol/ticket.re:3]: (feel free to resolve)

Comment (EduardoRFS) [reply protocol/ticket.re:3]: A lot of duplicated code and re exporting a type from Tezos_interop, it's probably a bad abstraction. Also nope, `Set_with_yojson_make` is completely different from `Map_with_yojson_make`.

Comment (EduardoRFS) [reply protocol/ticket.re:3]: It's already slightly improved on a future PR tho

**Pull request merged**